### PR TITLE
Passthrough potential errors for `EventLoopSender`

### PR DESCRIPTION
**Pull request by HyperHamster**
For this change I'm not sure if the error enum should be made `#[non_exhaustive]` or not, I'm simply not familiar enough with the codebase or your plans for the future to make that determination myself. I think it's fine that you chose to ignore these errors internally but it felt like bad practice to not pass through potential errors on a `pub` method.

### CODE REVIEW

File: alacritty_terminal/src/event_loop.rs [hunk 1]
```diff
@@ -339,13 +340,37 @@ impl event::Notify for Notifier {
...
+            EventLoopSendError::Send(x) => x.fmt(f),
+        }
+    }
+}
+
+impl std::error::Error for EventLoopSendError {
+    fn source(&self) -> Option<&(dyn std::error::Error + 'static)> {
+        match self {
+            EventLoopSendError::Io(x) => Some(x),
+            EventLoopSendError::Send(x) => Some(x),
```
Comment (chrisduerr) [inline alacritty_terminal/src/event_loop.rs:372]: `x` is not a good variable name.

File: alacritty_terminal/src/event_loop.rs [hunk 2]
```diff
@@ -339,13 +340,37 @@ impl event::Notify for Notifier {
...
+pub enum EventLoopSendError {
+    Io(io::Error),
+    Send(mpsc::SendError<Msg>),
+}
+
+impl Display for EventLoopSendError {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        match self {
+            EventLoopSendError::Io(x) => x.fmt(f),
+            EventLoopSendError::Send(x) => x.fmt(f),
```
Comment (chrisduerr) [inline alacritty_terminal/src/event_loop.rs:363]: `x` is not a good variable name.

**Review by chrisduerr — CHANGES_REQUESTED** [review]
I don't really mind this, but handling these errors is rarely ever useful.

File: alacritty_terminal/src/event_loop.rs (hunk 1 above)
Comment (HyperHamster) [reply alacritty_terminal/src/event_loop.rs]: Renamed to `err` to conform with other `std::error::Error` `impl`s in the project.

File: alacritty_terminal/src/event_loop.rs (hunk 2 above)
Comment (HyperHamster) [reply alacritty_terminal/src/event_loop.rs]: Renamed to `err` to conform with other `std::fmt::Display` `impl`s in the project.

**Review by kchibisov — APPROVED** [review]

**Review by chrisduerr — APPROVED** [review]

**Pull request merged**